tool/infra-testing-tool/go/cmd/testflow.go

```diff
@@ -0,0 +1,378 @@
+			startTestWG.Add(len(groupArray[g].Peers))
+			// iterate over peers in group
+			for k := range groupArray[g].Peers {
+				groupIndex := g
+				testIndex := j
+				peerIndex := k
+				go func(newTestWG, startTestWG *sync.WaitGroup) {
+					// create new test with correct variables
+					_, err = groupArray[groupIndex].Peers[peerIndex].P.NewTest(ctx, &server.NewTest_Request{
+						GroupName: groupArray[groupIndex].Name,
+						TestN:     int64(testIndex),
+						Type:      groupArray[groupIndex].Tests[testIndex].TypeInternal,
+						Size:      int64(groupArray[groupIndex].Tests[testIndex].SizeInternal),
+						Interval:  int64(groupArray[groupIndex].Tests[testIndex].IntervalInternal),
+						Amount:    int64(groupArray[groupIndex].Tests[testIndex].AmountInternal),
+					})
+					logging.Log(fmt.Sprintf("added test - test: '%v' in group: '%v' on node: %v", testIndex, groupArray[groupIndex].Name, groupArray[g].Peers[peerIndex].Tags[aws.Ec2TagName]))
+
+					if err != nil {
+						logging.Log(err.Error())
+					}
+
+					newTestWG.Done()
+					// makes sure all tests are synced up
+					newTestWG.Wait()
+
+					// start said test
+					_, err = groupArray[groupIndex].Peers[peerIndex].P.StartTest(ctx, &server.StartTest_Request{
+						GroupName: groupArray[groupIndex].Name,
+						TestN:     int64(testIndex),
+					})
+					if err != nil {
+						logging.Log(err.Error())
+					}
+					time.Sleep(time.Second * 3)
+
+					logging.Log(fmt.Sprintf("started test - test: '%v' in group: '%v' on node: %v", testIndex, groupArray[g].Name, groupArray[g].Peers[peerIndex].Tags[aws.Ec2TagName]))
+
+					startTestWG.Done()
+				}(&newTestWG, &startTestWG)
+			}
+		}
+
+		startTestWG.Wait()
+
+		var wg sync.WaitGroup
+		for j := range groupArray[g].Peers {
+			for k := range groupArray[g].Tests {
+				wg.Add(1)
+
+				peerIndex := j
+				testIndex := k
+				go func() {
+					defer wg.Done()
+
+					var err error
+					var isRunning = true
+
+					// loops until `TestIsRunning` return res,s false
+					for isRunning && err == nil {
+						time.Sleep(time.Second * 3)
+
+						var res *server.IsTestRunning_Response
+						res, err = groupArray[g].Peers[peerIndex].P.IsTestRunning(ctx, &server.IsTestRunning_Request{
+							GroupName: groupArray[g].Name,
+							TestN:     int64(testIndex),
+						})
+
+						if err == nil {
+							isRunning = res.TestIsRunning
+						}
+					}
+
+					if !isRunning {
+						logging.Log(fmt.Sprintf("test finished - test: '%v' in group: '%v' on node: %v", testIndex, groupArray[g].Name, groupArray[g].Peers[peerIndex].Tags[aws.Ec2TagName]))
+					} else {
+						logging.Log(fmt.Sprintf("test error - test: '%v' in group: '%v' on node: %v: %s",
+							testIndex, groupArray[g].Name, groupArray[g].Peers[peerIndex].Tags[aws.Ec2TagName], err.Error()))
+					}
+				}()
+			}
+		}
+
+		logging.Log("waiting for all tests to finish ...")
+
+		// wait for wait group to finish
+		wg.Wait()
+
+	}
+
+	logging.Log("all tests are finished")
```

Comment:
This log and the log above line 368 are misleading. Something like "waiting for gRPC calls to be sent..." and "all gRPC calls have been sent" should be better.

Reply:
To make it clearer for the user, you could also add an info like: "peers are still running the requested tests... ctrl + c to stop the tests".